dkpro-core-io-conll-asl/src/test/java/org/dkpro/core/io/conll/ConllUReaderTest.java

```diff
@@ -75,6 +84,66 @@ public void test()
         assertMorph(morphologicalFeeatures, select(jcas, MorphologicalFeatures.class));
     }
 
+    @Test
+    public void testDocumentID()
+            throws Exception
+    {
+        CollectionReaderDescription reader = createReaderDescription(
+                ConllUReader.class,
+                ConllUReader.PARAM_LANGUAGE, "en",
+                ConllUReader.PARAM_SOURCE_LOCATION, "src/test/resources/conll/u_v2/",
+                ConllUReader.PARAM_PATTERNS, "conllu-paragraph_and_document_boundaries.conll");
+
+        JCas jcas = new JCasIterable(reader).iterator().next();
+
+        AnnotationIndex<DocumentMetaData> index = jcas.getAnnotationIndex(DocumentMetaData.class);
+        DocumentMetaData m = index.iterator().get();
+        final String actualDocumentID = m.getDocumentId();
+        final String expectedDocumentID = "mf920901-001";
+
+        Assert.assertEquals("Document ID mismatch", expectedDocumentID, actualDocumentID);
+    }
+
+    @Test
+    public void testParagraphs()
+            throws Exception
+    {
+        CollectionReaderDescription reader = createReaderDescription(
+                ConllUReader.class,
+                ConllUReader.PARAM_LANGUAGE, "en",
+                ConllUReader.PARAM_SOURCE_LOCATION, "src/test/resources/conll/u_v2/",
+                ConllUReader.PARAM_PATTERNS, "conllu-multiple_paragraphs.conll");
+
+        JCas jcas = new JCasIterable(reader).iterator().next();
+
+        AnnotationIndex<Paragraph> index = jcas.getAnnotationIndex(Paragraph.class);
+
+        List<String> paragraphIDs = new LinkedList<>();
```

Comment:
Normally, ArrayList should be used unless there is a specific reason to use another list implementation.